go/libkb/keyfamily.go

```diff
@@ -579,14 +586,24 @@ func (cki *ComputedKeyInfos) Delegate(kid keybase1.KID, tm *KeybaseTime, sigid k
 	return
 }
 
-// DelegateSharedDHKey inserts the new shared DH public key into the
-// list of known generations of DH public keys.
-func (cki *ComputedKeyInfos) DelegateSharedDHKey(s *SharedDHKeyChainLink) (err error) {
-	cki.SharedDHKeys[s.generation] = keybase1.SharedDHKey{
-		Gen:   int(s.generation),
-		Kid:   s.GetDelegatedKid(),
-		Seqno: int(s.GetSeqno()),
+// DelegatePerUserKey inserts the new per-user key into the list of known per-user keys.
+func (cki *ComputedKeyInfos) DelegatePerUserKey(s *PerUserKeyChainLink) (err error) {
+	entry := keybase1.PerUserKey{
+		Gen:    int(s.generation),
+		Seqno:  int(s.GetSeqno()),
+		SigKID: s.sigKID,
+		EncKID: s.encKID,
+	}
+	if entry.Gen <= 0 {
+		return fmt.Errorf("invalid per-user-key generation %v", entry.Gen)
+	}
+	if entry.SigKID.IsNil() {
+		return errors.New("nil per-user-key sig kid")
+	}
+	if entry.EncKID.IsNil() {
```

Comment:
Should the chain link have failed to parse earlier, if it's going to fail any of these checks?